### ISSUE TITLE
experimental ipvlan : kernel panic during container stop with ipvlan driver

### ISSUE THREAD

**Issue by mavenugo** [docker/libnetwork#1005]
```
$ sudo docker network create -d ipvlan iv
eb00df4e38c04108d61c3b1a8ed52679851da16ffcba191de965aa6d282b8063

$ sudo docker run --net=iv --name=ic1 -d busybox top
f8268ef608c5ca7000f202dadb0b3b75dc5d95887ea72872eac95e60beee1651

$ sudo docker stop ic1
```

Results in :

```
[  220.799094] general protection fault: 0000 [#1] SMP
[  221.500878] Modules linked in: xt_nat dummy ipvlan xt_addrtype xt_conntrack br_netfilter overlay xt_CHECKSUM iptable_mangle ipt_MASQUERADE nf_nat_masquerade_ipv4 iptable_nat nf_conntrack_ipv4 nf_defrag_ipv4 nf_nat_ipv4 nf_nat nf_conntrack xt_tcpudp bridge stp llc iptable_filter vboxsf(OE) ip_tables x_tables ppdev vboxvideo(OE) drm parport_pc parport 8250_fintek mac_hid vboxguest(OE) i2c_piix4 sunrpc autofs4 crct10dif_pclmul crc32_pclmul ghash_clmulni_intel aesni_intel ahci aes_x86_64 lrw gf128mul glue_helper ablk_helper cryptd libahci video e1000
[  225.551733] CPU: 1 PID: 1282 Comm: docker-1.11.0-d Tainted: G           OE  3.19.0-25-generic #26-Ubuntu
[  226.773738] Hardware name: innotek GmbH VirtualBox/VirtualBox, BIOS VirtualBox 12/01/2006
[  226.997284] task: ffff8800daca4e80 ti: ffff880035da8000 task.ti: ffff880035da8000
[  227.178937] RIP: 0010:[<ffffffffc04c05e0>]  [<ffffffffc04c05e0>] ipvlan_ht_addr_del+0x10/0x40 [ipvlan]
[  227.403816] RSP: 0018:ffff880035dab660  EFLAGS: 00010286
[  227.777243] RAX: ffff880119868cd8 RBX: ffff88011a3ab000 RCX: 000000000e0eebfc
[  227.957517] RDX: dead000000200200 RSI: 0000000000000001 RDI: ffff880119868cc0
[  228.275854] RBP: ffff880035dab698 R08: ffffffff81d28ec8 R09: ffffffffc04c41a0
[  228.883721] R10: 0000000000000292 R11: 0000000000000020 R12: ffff88011a3ab8c0
[  229.056311] R13: ffff880119868cc0 R14: ffff8800361d1000 R15: 0000000000000000
[  230.571288] FS:  00007f14910ef700(0000) GS:ffff88011fc80000(0000) knlGS:0000000000000000
[  231.231957] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  231.810098] CR2: 000000c820911000 CR3: 0000000119863000 CR4: 00000000000406e0
[  231.959174] DR0: 0000000000000000 DR1: 0000000000000000 DR2: 0000000000000000
[  232.271878] DR3: 0000000000000000 DR6: 00000000fffe0ff0 DR7: 0000000000000400
[  232.830785] Stack:
[  232.885553]  ffffffffc04c1cc5 ffff880035dab678 020013acc03441bc ffff880035dab698
[  233.050997]  00000000fffffffd 0000000000000000 0000000000000002 ffff880035dab6d8
[  233.478795]  ffffffff810968df ffff8800d7285500 ffffffff81cebec0 0000000000000002
[  233.847465] Call Trace:
[  233.854995]  [<ffffffffc04c1cc5>] ? ipvlan_addr4_event+0x85/0x210 [ipvlan]
[  233.867562]  [<ffffffff810968df>] notifier_call_chain+0x4f/0x80
[  234.118220]  [<ffffffff81096c3b>] __blocking_notifier_call_chain+0x4b/0x70
[  234.808210]  [<ffffffff81096c76>] blocking_notifier_call_chain+0x16/0x20
[  234.815863]  [<ffffffff81733323>] __inet_del_ifa+0x153/0x2b0
[  235.456241]  [<ffffffff81734b32>] inetdev_event+0x182/0x540
[  236.150137]  [<ffffffff816f1605>] ? nlmsg_notify+0xd5/0xf0
[  236.259528]  [<ffffffff810968df>] notifier_call_chain+0x4f/0x80
[  236.436483]  [<ffffffff81096a16>] raw_notifier_call_chain+0x16/0x20
[  237.468577]  [<ffffffff816ba6d9>] call_netdevice_notifiers_info+0x39/0x70
[  237.636747]  [<ffffffff816be663>] dev_change_net_namespace+0xc3/0x3f0
[  238.011323]  [<ffffffff81314398>] ? security_capable+0x18/0x20
[  239.491918]  [<ffffffff8107f75d>] ? ns_capable+0x2d/0x60
[  240.153072]  [<ffffffff816d2c46>] do_setlink+0xa6/0xb80
[  240.297178]  [<ffffffff813e1049>] ? nla_parse+0xb9/0x120
[  240.405040]  [<ffffffff816d37e9>] rtnl_setlink+0xc9/0x140
[  240.523100]  [<ffffffff816cffc2>] rtnetlink_rcv_msg+0x92/0x260
[  241.255479]  [<ffffffff813cb616>] ? rhashtable_lookup_compare+0x36/0x70
[  241.609542]  [<ffffffff816ed5fe>] ? __netlink_lookup+0x3e/0x50
[  241.610404]  [<ffffffff816cff30>] ? rtnetlink_rcv+0x40/0x40
[  241.908920]  [<ffffffff816f1509>] netlink_rcv_skb+0xb9/0xe0
[  242.024237]  [<ffffffff816cff1c>] rtnetlink_rcv+0x2c/0x40
[  242.275279]  [<ffffffff816f0c41>] netlink_unicast+0x111/0x1b0
[  243.497753]  [<ffffffff816f0fe0>] netlink_sendmsg+0x300/0x650
[  244.012428]  [<ffffffff81360430>] ? aa_sk_perm.isra.4+0x70/0x170
[  244.122775]  [<ffffffff816a540c>] do_sock_sendmsg+0x8c/0x100
[  244.250827]  [<ffffffff813cba1d>] ? rhashtable_insert+0x4d/0x80
[  245.493149]  [<ffffffff816edb4a>] ? netlink_insert+0x9a/0xc0
[  245.714432]  [<ffffffff81213585>] ? __fget_light+0x25/0x70
[  245.861513]  [<ffffffff812135e3>] ? __fdget+0x13/0x20
[  246.856958]  [<ffffffff816a56b2>] SYSC_sendto+0x1a2/0x210
[  246.999056]  [<ffffffff816a1ed7>] ? sock_alloc_file+0x97/0x110
[  247.469417]  [<ffffffff81213c3d>] ? __fd_install+0x4d/0x70
[  247.866390]  [<ffffffff81213c85>] ? fd_install+0x25/0x30
[  248.006597]  [<ffffffff816a5ebe>] SyS_sendto+0xe/0x10
[  248.210825]  [<ffffffff817cb5cd>] system_call_fastpath+0x16/0x1b
[  248.786318] Code: 44 00 00 48 8d 7e 08 e8 0f fd ff ff eb c1 0f 1f 00 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 48 8b 47 18 48 8b 57 20 48 85 c0 <48> 89 02 74 04 48 89 50 08 48 b8 00 02 20 00 00 00 ad de 40 84
[  250.909896] RIP  [<ffffffffc04c05e0>] ipvlan_ht_addr_del+0x10/0x40 [ipvlan]
[  251.564303]  RSP <ffff880035dab660>
[  251.843680] Clocksource tsc unstable (delta = 28121827592 ns)
[  251.984036] ---[ end trace ca21570c2fb7cc22 ]---

```

**Comment by mavenugo** [comment]
ping @nerdalert

**Comment by nerdalert** [comment]
Thanks @mavenugo This is also related to kernel version 3.19, the first kernel rev that included Ipvlan support. Added an example tests and output in issue #1005 that is also affected by the kernel 3.19 bugs.

**Comment by nerdalert** [comment]
This was resolved by PR #1011 by prohibiting the early Ipvlan implementations with bugs in < kernel 4.0.0
